sealtk/core/test/ImageUtils.cpp

```diff
@@ -0,0 +1,340 @@
+{
+  m_surface.create();
+  QVERIFY(m_context.create());
+  QVERIFY(m_context.makeCurrent(&m_surface));
+
+  m_fbo.reset(new QOpenGLFramebufferObject{FBO_SIZE, FBO_SIZE});
+  QVERIFY(m_fbo->isValid());
+
+  m_pixelPackedShader.reset(new QOpenGLShaderProgram);
+  QVERIFY(m_pixelPackedShader->create());
+  QVERIFY(m_pixelPackedShader->addShaderFromSourceCode(
+    QOpenGLShader::Vertex, VERTEX_SHADER_CODE));
+  QVERIFY(m_pixelPackedShader->addShaderFromSourceCode(
+    QOpenGLShader::Fragment, PIXEL_PACKED_FRAGMENT_SHADER_CODE));
+  QVERIFY(m_pixelPackedShader->link());
+
+  m_planePackedShader.reset(new QOpenGLShaderProgram);
+  QVERIFY(m_planePackedShader->create());
+  QVERIFY(m_planePackedShader->addShaderFromSourceCode(
+    QOpenGLShader::Vertex, VERTEX_SHADER_CODE));
+  QVERIFY(m_planePackedShader->addShaderFromSourceCode(
+    QOpenGLShader::Fragment, PLANE_PACKED_FRAGMENT_SHADER_CODE));
+  QVERIFY(m_planePackedShader->link());
+
+  float vertexData[] = {
+    0.0f, 0.0f, 0.0f,
+    1.0f, 0.0f, 0.0f,
+    1.0f, 1.0f, 0.0f,
+    0.0f, 1.0f, 0.0f
+  };
+
+  m_vertices.create();
+  m_vertices.bind();
+  m_vertices.allocate(vertexData, static_cast<int>(sizeof(vertexData)));
+  m_vertices.release();
+
+  m_context.doneCurrent();
+}
+
+// ----------------------------------------------------------------------------
+void TestImageUtils::cleanupTestCase()
+{
+  QVERIFY(m_context.makeCurrent(&m_surface));
+  m_fbo.reset();
+  m_pixelPackedShader.reset();
+  m_planePackedShader.reset();
+  m_context.doneCurrent();
+  m_surface.destroy();
+}
+
+// ----------------------------------------------------------------------------
+void TestImageUtils::imageToTexture()
+{
+  // Get test parameters
+  QFETCH(size_t, channels);
+  QFETCH(ptrdiff_t, xStride);
+  QFETCH(ptrdiff_t, yStride);
+  QFETCH(ptrdiff_t, cStride);
+
+  // Create input image
+  auto const& image = makeTestImage(channels, xStride, yStride, cStride);
+  auto imageContainer =
+    std::make_shared<kv::simple_image_container>(image);
+
+  // Bind FBO
+  QVERIFY(m_context.makeCurrent(&m_surface));
+  QVERIFY(m_fbo->bind());
+
+
+  // Create and fill texture
+  QOpenGLTexture texture{QOpenGLTexture::Target2DArray};
+  ::sealtk::core::imageToTexture(texture, imageContainer);
+
+  // Select shader
+  auto const& shader =
+    (texture.layers() > 1 ? m_planePackedShader : m_pixelPackedShader);
+
+  // Render to FBO
+  m_context.functions()->glViewport(0, 0, FBO_SIZE, FBO_SIZE);
+  m_context.functions()->glClearColor(0.0f, 0.0f, 0.0f, 0.0f);
+  m_context.functions()->glClear(GL_COLOR_BUFFER_BIT);
+
+  shader->bind();
+  m_vertices.bind();
+  texture.bind();
+
+  shader->setAttributeBuffer(0, GL_FLOAT, 0, 3);
+  shader->enableAttributeArray(0);
+
+  m_context.functions()->glDrawArrays(GL_TRIANGLE_FAN, 0, 4);
+
+  m_vertices.release();
+  texture.release();
+  shader->release();
+
+  // Get FBO contents
+  m_fbo->release();
+  auto const& out = m_fbo->toImage();
+
+  /* TESTING
```

Comment:
Please remove the stale testing code.

Reply:
I noticed that, and decided to leave it, as it's likely to be very helpful if anything goes wrong and someone needs to debug the test. I can remove it if you feel really strongly about it, though. (Alternatively, I could improve the comment to explain what it does and how to use it.)

Reply:
In that case, I think the best thing to do would be to wrap it in an `#ifdef` or `#if 0` so that it's clear that it is deliberate and not stale. Please also add the explanation.